### PR TITLE
Postman captures interactions

### PR DESCRIPTION
**Pull request by niclim**
## 🍗 Description
_What does this PR do? Anything folks should know?_

## 📚 References
_Links to relevant docs (Notion, Twist, GH issues, etc.), if applicable._

## 👹 QA
_How can other humans verify that this PR is correct?_

### CODE REVIEW

File: projects/optic/src/commands/oas/captures/streams/sources/postman.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export type PostmanEntry = {
+  // TODO specify the postman collection type here
+};
+
+export interface PostmanCollectionEntries extends AsyncIterable<PostmanEntry> {}
+export interface TryPostmanCollection
+  extends AsyncIterable<Result<PostmanEntry, Error>> {}
+
+export class PostmanCollectionEntries {
+  static async *fromReadable(source: Readable): TryPostmanCollection {
```
Comment (acunniffe) [inline projects/optic/src/commands/oas/captures/streams/sources/postman.ts:17]: collection bytes -> an iterable postman req/res pairs

File: projects/optic/src/commands/oas/captures/interaction.ts [hunk 2]
```diff
@@ -150,6 +151,14 @@ export class CapturedInteraction {
       },
     };
   }
+
+  static fromPostmanCollection(
+    postmanEntry: PostmanEntry
+  ): CapturedInteraction {
+    // TODO convert postman entry to interaction
```
Comment (acunniffe) [inline projects/optic/src/commands/oas/captures/interaction.ts:158]: Convert whatever the postman req/res looks like into a CapturedInteraction

**Comment by kevinswiber** [comment]
@acunniffe @niclim Thanks for this!  Question for you... So Postman Collections have a Request model.  The Request model contains lots of HTTP request information, such as method, scheme, authority, path, query parameters, headers, and a request body.  Each Request optionally has an array of Examples.  Examples are Request-Response pairs.  Maybe a bulleted list can show this better.

* Request
  * method
  * URL
  * headers
  * body
  * Examples[]*
    * ExampleRequest
      * method
      * URL
      * headers
      * body
    * ExampleResponse
      * status code
      * headers
      * body

Now, each Request, without Examples, will have information that can be validated.  Would it be possible to validate that information without having a corresponding response?

**Comment by acunniffe** [comment]
@kevinswiber I think so -- we're looking into what change we'd have to make to the types / our code to support that. We'll get back to you tomorrow. 

This bullet list is super helpful! I get it now. 

Basically if you had 1 request, with 5 examples, then you'd want to emit 5 `CapturedInteraction` (one for each pair). If you had 1 request with no examples you'd want to emit 1 `CapturedInteraction` without a response.

**Comment by kevinswiber** [comment]
@acunniffe Cool.  Let me know.  I made a PR to this draft PR.  I actually don't know the etiquette here.  Seems like a regular PR on top of a draft PR makes sense more than draft-squared, but this is new territory for me.

https://github.com/opticdev/optic/pull/1802

**Comment by niclim** [comment]
> @acunniffe Cool. Let me know. I made a PR to this draft PR. I actually don't know the etiquette here. Seems like a regular PR on top of a draft PR makes sense more than draft-squared, but this is new territory for me.
> 
> #1802

PR looks good! Thanks for the contribution, let me know when it's good to go from your side and I can merge it in here and get it through our checks / code + released!

Also updated the `CapturedInteraction` type to allow for nullable responses [2536b76](https://github.com/opticdev/optic/pull/1796/commits/2536b76adfbda90ccab2cc34a18592451109a4a7)

**Comment by kevinswiber** [comment]
Thanks, all.  With the review approved on #1802, I think it's ready to be merged into this branch.

**Comment by niclim** [comment]
`0.42.3-0` is out - `npm i -g @useoptic/optic@0.42.3-0` to test, will be doing some testing before releasing a patch version!

**Pull request merged**